tests/pass-dep/libc/libc-epoll.rs

```diff
@@ -261,24 +259,6 @@ fn test_epoll_eventfd() {
     check_epoll_wait::<8>(epfd, &[(expected_event, expected_value)]);
 }
 
-fn test_pointer() {
-    // Create an epoll instance.
-    let epfd = unsafe { libc::epoll_create1(0) };
-    assert_ne!(epfd, -1);
-
-    // Create a socketpair instance.
-    let mut fds = [-1, -1];
-    let res = unsafe { libc::socketpair(libc::AF_UNIX, libc::SOCK_STREAM, 0, fds.as_mut_ptr()) };
-    assert_eq!(res, 0);
-
-    // Register fd[1] with EPOLLIN|EPOLLOUT|EPOLLET
-    let data = MaybeUninit::<u64>::uninit().as_ptr();
-    let mut ev =
-        libc::epoll_event { events: EPOLL_IN_OUT_ET, u64: data.expose_provenance() as u64 };
-    let res = unsafe { libc::epoll_ctl(epfd, libc::EPOLL_CTL_ADD, fds[1], &mut ev) };
-    assert_eq!(res, 0);
-}
```

Comment:
@tiif what is the point of this test? `data.expose_provenance()` is a completely normal integer, so this isn't testing anything that the other tests wouldn't also cover.

Reply:
We can already get it back out now though, without blocking epoll_wait.

And even then I don't get the point of the test. It's just testing that we're passing through the integer? We're already testing that in the other tests...

Reply:
I assumed we'd want to test pointers in a way that people would use this API, even though it is broken

Reply:
:shrug: seems rather redundant to me since we already test int2ptr casts and the u64 data behavior separately and there's no reason to think there could be any interesting interactions here. So having to maintain more tests is a waste of time IMO.

Reply:
The one case that I think could be interesting is passing epoll_ctl a buffer that actually stores a pointer there. This requires some shenanigans, like defining an `epoll_event_ptr` type and then casting the `*mut epoll_event_ptr` to `*mut epoll_event` and passing that to `epoll_ctl`. There would be no `expose_provenance`. I don't know if the current implementation would preserve the provenance or not in that case.

Reply:
Yea, that seems better. It's just a libc crate limitation, not something of the libc API/kernel API